hypershift-operator/controllers/nodepool/metrics/metrics.go

```diff
@@ -71,6 +78,22 @@ func RecordNodePoolAvailableReplicas(nodePool *hyperv1.NodePool) {
 	nodePoolAvailableReplicas.With(labels(nodePool)).Set(float64(nodePool.Status.Replicas))
 }
 
+func RecordNodePoolInfo(nodePool *hyperv1.NodePool, hcluster *hyperv1.HostedCluster) {
+	labels := labels(nodePool)
+
+	// Fill the extra info there
+	npPlatform := nodePool.Spec.Platform
+
+	labels["_id"] = hcluster.Spec.ClusterID
+	if npPlatform.Type == hyperv1.AWSPlatform && npPlatform.AWS != nil {
+		labels["ec2_instance_type"] = npPlatform.AWS.InstanceType
+	} else {
+		labels["ec2_instance_type"] = ""
+	}
```

Comment:
**Question:** Is there any point even setting this label if the platform isn't AWS?

Reply:
Yes, the `hypershift` operator will crash loop if this label is not set.

Reply:
Understood, would the metric then contain an `ec2_instance_type` or does golang prometheus ditch the label if it's empty? 

Also would it make sense to change this label's name to `instance_type` and have another label for the platform? This label could then be re-used for other platforms, as `ec2` is aws specific.

Reply:
> Understood, would the metric series then contain an ec2_instance_type = ""

Yes 👍 

> does golang prometheus ditch the label if it's empty?

Nope

> Also would it make sense to change this label's name to instance_type

This is the name I wanted to use at the beginning (see https://github.com/openshift/hypershift/commit/3f8ea9c549b4d48661924ed806e9d395fa0d464d), but on a second thought I preferred being more explicit and indicate that this label is for AWS only. Indeed other platforms use a different naming ("machine types" for GCP, "VM size" on Azure), also on GCP it is even possible to fine tune the number of cores & the memory in which case the machine type does not make sense anymore.

Eventually I could rename the label to `AWS_instance_type` to be more clear on the platform provider.